PoseEstimation/poseEstimation_main.py

```diff
@@ -0,0 +1,171 @@
+import os, requests
+import sys
+import numpy as np
+import pandas as pd
+import h5py
+import typing
+from pathlib import Path
+import matplotlib.pyplot as plt
+from tifffile import imwrite, TiffWriter, TiffFile
+import yaml
+import cv2
+import copy
+from datetime import datetime
+import glob
+
+sys.path.append("..")
+import utilities as utils
+import definitions as defs
+import poseEstimation_parameters as poseEst_params
+import poseEstimation_definitions as poseEst_defs
+
+import deeplabcut
+print("Imported DLC!", flush=True)
+
+# Import for PyInstaller
+from multiprocessing import freeze_support
+freeze_support()
+
+def main(poseEstimation_params: poseEst_params.PoseEstimationParameters):
+
+    """
+    DeepLabCut algorithm
+    """
+    # --------------- Read danse parameters ---------------
+    operations: str      = poseEstimation_params.params[defs.DoricFile.Attribute.Group.OPERATIONS]
+    filepath: str        = poseEstimation_params.paths[defs.Parameters.Path.FILEPATH]
+    datapath: str        = poseEstimation_params.paths[defs.Parameters.Path.H5PATH]
+    project_folder: str  = poseEstimation_params.params[poseEst_defs.Parameters.danse.PROJECT_FOLDER]
+    bodypart_names       = poseEstimation_params.params[poseEst_defs.Parameters.danse.BODY_PART_NAMES].split(', ')
+    bodypart_colors      = poseEstimation_params.params[poseEst_defs.Parameters.danse.BODY_PART_COLORS].split(', ')
+    extracted_frames     = poseEstimation_params.params[poseEst_defs.Parameters.danse.EXTRACTED_FRAMES]
+
+    [file_, video_path, path_config_file, training_coordinates] = create_project(filepath, datapath, project_folder, bodypart_names, extracted_frames, poseEstimation_params)
```

Comment:
Do we have to keep the `file_` open? Can we open and close it in the `create_project`, and then open and close in `save_coords_to_doric`?

Reply:
I don't think we need anything but `video_path, path_config_file`